coretex/cli/commands/project.py

```diff
@@ -20,23 +20,70 @@
 import click
 
 from ..modules import ui, project_utils, utils, user
-from ...entities import Project
-from ...networking import NetworkRequestError
-from ...configuration import loadConfig, saveConfig
+from ...entities import Project, ProjectType, ProjectVisibility
+from ...networking import NetworkRequestError, EntityNotCreated, networkManager, RequestFailedError
+from ...configuration import loadConfig
 
 
-def selectProject(projectId: int) -> None:
+@click.command()
+@click.option("--name", "-n", type = str, help = "Project name")
+@click.option("--type", "-t", type = int, help = "Project type")
+@click.option("--description", "-d", type = str, help = "Project description")
+def create(name: Optional[str], type: Optional[int], description: Optional[str], visibility: Optional[int]) -> None:
+    if name is None:
+        name = ui.clickPrompt("Please enter name of the project you want to create:", type = str)
+
+    if type is None:
+        type = project_utils.selectProjectType()
+
+    if description is None:
+        description = ui.clickPrompt("Please enter your project's description:", type = str)
+
+    try:
+        project = Project.createProject(name, ProjectType(type), description = description)
+        ui.successEcho(f"Project \"{name}\" created successfully.")
+        selectNewProject = ui.clickPrompt("Do you want to select new project as default? (Y/n)", type = bool, default = True)
+
+        if selectNewProject:
+            project_utils.selectProject(project.id)
+            ui.successEcho(f"Project \"{project.name}\" successfully selected.")
+    except EntityNotCreated:
+        raise click.ClickException(f"Failed to create project \"{name}\".")
+
+
+@click.command()
+@click.option("--name", "-n", type = str, help = "Project name")
+@click.option("--description", "-d", type = str, help = "Project description")
+def edit(name: Optional[str], description: Optional[str], visibility: Optional[int]) -> None:
     config = loadConfig()
-    config["projectId"] = projectId
-    saveConfig(config)
+    defaultProjectId = config.get("projectId")
+    if defaultProjectId is None and name is None:
+        ui.errorEcho(f"To use edit command you need to specifiy project name using \"--project\" or \"-p\" flag, or you can select default project using \"coretex project select\" command.")
+
+    if name is None and defaultProjectId is not None:
+        selectedProject = Project.fetchById(defaultProjectId)
+    else:
+        selectedProject = Project.fetchOne(name = name)
+
+    if description is None:
+        description = ui.clickPrompt("Please enter new description for your project", type = str, default = selectedProject.description)
```

Comment:
Descriptions can be quite large so let's not display the default value of the description